src/Lecture2.hs

```diff
@@ -1,170 +1,277 @@
-{- |
-Module                  : Lecture2
-Copyright               : (c) 2021-2022 Haskell Beginners 2022 Course
-SPDX-License-Identifier : MPL-2.0
-Maintainer              : Haskell Beginners 2022 Course <haskell.beginners2022@gmail.com>
-Stability               : Stable
-Portability             : Portable
-
-Exercises for the Lecture 2 of the Haskell Beginners course.
-
-As in the previous section, implement functions and provide type
-signatures. If the type signature is not already written, write the
-most polymorphic type signature you can.
-
-Unlike exercises to Lecture 1, this module also contains more
-challenging exercises. You don't need to solve them to finish the
-course but you can if you like challenges :)
--}
-
+-- |
+-- Module                  : Lecture2
+-- Copyright               : (c) 2021-2022 Haskell Beginners 2022 Course
+-- SPDX-License-Identifier : MPL-2.0
+-- Maintainer              : Haskell Beginners 2022 Course <haskell.beginners2022@gmail.com>
+-- Stability               : Stable
+-- Portability             : Portable
```

Comment:
Hey @mcdiarmids 👋🏻 

I think that you've probably used `ormolu` to format the code. I still can review your solutions this time but it makes it difficult to review the diff when the entire file content has changed. So if you can avoid using changing the comments with autoformatting next time I would greatly appreciate that 🙂

Reply:
Sorry about that! Will keep that in mind for the remaining exercises.